### PR TITLE
Add some debugging and error checking to the scripts

### PR DESCRIPTION
**Pull request by agbrook**
While trying to debug a separate issue with permissions on d42, I found that I wasn't getting enough data from the script to get started. This led me to making these changes to the script to handle error codes from d42 properly and putting them in the log. Also, adding some debug logging to get more information from the salt debug logs if that were necessary. 

This pull request also adds in a setting in the settings file to determine if the SSL certificate should be checked. In my environment, I want the certificate to be checked, but I know not all of environments need that. This setting should be clear in the settings file instead of having to modify the script to change that.

### CODE REVIEW

File: api_test.py [hunk 1]
```diff
@@ -83,28 +90,37 @@ def generate_simple_query(fields, nodename):
 def main(): 
 	
 	config = get_config('settings_d42.yaml')
-	nodename = 'ubuntu.saltmaster5'
+	if len(sys.argv) < 2:
```
Comment (nullbuddy1243) [inline api_test.py:93]: Ah nice find: in the api_test file I had left the nodename to a default used in development.

**Review by nullbuddy1243 — APPROVED** [review]
Thanks for the changes!  Definitely helpful to have a bit more debugging facilities.

**Comment by nullbuddy1243** [comment]
Hey @agbrook!  I really appreciate your pull request and am glad to approve and merge.  

By the way, I'd love to know what you've been using / planning on using this integration for.  It's still on the new side so feedback is very welcome. 

Best,
Will

**Comment by agbrook** [comment]
I'm planning on using it as you've mentioned in the blog and the README.
I'd like to start referencing servers based on service_level when running
salt states, specifically when running and installing patches. However,
I've been thinking of using it store some pillar information in custom
fields, so that I don't have to store them on the filesystem. This will
help out greatly when building servers. I can put all the information in
device42, then use that information later when the server is being
built/installed. This way some of the more sensitive stuff is on an
audited/authenticated system external to salt where the backups are
encrypted.

-Andy

**Comment by nullbuddy1243** [comment]
@agbrook Awesome.  I'm glad it is helpful to you!  Definitely let me know how your time with it goes or if you come up with any suggestions for extensions!

**Pull request merged**